### ISSUE TITLE
The contact name is not recognized

### ISSUE THREAD

**Issue by marmistrz** [moezbhatti/qksms#655]
It's not really easy to reproduced, happens only with one contact. Their phone number is in format +48XXXXXXXXX. QKSMS doesn't display the contact name but uses `48XXXXXXXXX` instead.

Logcat here:

```
10-05 21:40:02.316  4282  4282 I TriggerService: comparePriorities called
10-05 21:40:08.321  4282  4282 I TriggerBroadcastReceiver: Broadcast received: android.intent.action.BATTERY_CHANGED
10-05 21:40:08.321  4282  4282 I TriggerService: batterylevel changed to 20
10-05 21:40:08.322  4282  4282 I TriggerService: compareTriggers called
10-05 21:40:08.322  4282  4282 I TriggerService: comparePriorities called
10-05 21:40:09.882   547  1563 D audio_hw_primary: out_set_parameters: enter: usecase(1: low-latency-playback) kvpairs: routing=2
10-05 21:40:09.883   547  1563 D audio_hw_extn: audio_extn_set_anc_parameters: anc_enabled:0
--------- beginning of system
10-05 21:40:09.885  1359  2539 I ActivityManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.moez.QKSMS/.ui.MainActivity-Default bnds=[13,1322][224,1630] (has extras)} from uid 10027 on display 0
10-05 21:40:09.894   547  1563 D audio_hw_primary: start_output_stream: enter: stream(0xf2f3c380)usecase(1: low-latency-playback) devices(0x2)
10-05 21:40:09.908   547  1563 D audio_hw_primary: select_devices: out_snd_device(2: speaker) in_snd_device(0: )
10-05 21:40:09.908   547  1563 I msm8974_platform: platform_check_and_set_codec_backend_cfg: Usecase bitwidth 16, samplerate 48000
10-05 21:40:09.908   547  1563 I msm8974_platform: platform_check_and_set_codec_backend_cfg: new_snd_devices[0] is 2
10-05 21:40:09.908   547  1563 I msm8974_platform: platform_check_codec_backend_cfg Codec selected backend: 0 current bit width: 16 and sample rate: 48000
10-05 21:40:09.908   547  1563 D msm8974_platform: platform_check_codec_backend_cfg: resetting sample_rate back to default, backend_idx: 0
10-05 21:40:09.908   547  1563 I msm8974_platform: platform_check_codec_backend_cfg Codec selected backend: 0 updated bit width: 16 and sample rate: 48000
10-05 21:40:09.908   547  1563 D hardware_info: hw_info_append_hw_type : device_name = speaker
10-05 21:40:09.908   547  1563 D audio_hw_primary: enable_snd_device: snd_device(2: speaker)
10-05 21:40:09.908   547  1563 I soundtrigger: audio_extn_sound_trigger_update_device_status: device 0x2 of type 0 for Event 1, with Raise=0
10-05 21:40:09.908   547  1563 D aud_dev_arbi: acquireOneDevice: device owner for device 2 is 1
10-05 21:40:09.913  1359  1851 I ActivityManager: Start proc 24643:com.moez.QKSMS/u0a82 for activity com.moez.QKSMS/.ui.MainActivity-Default
10-05 21:40:09.914   547  1563 I audio_hw_primary: select_devices Selected apptype: 69937
10-05 21:40:09.914   547  1563 D soundtrigger: audio_extn_sound_trigger_update_stream_status: uc_id 1 of type 0 for Event 3, with Raise=0
10-05 21:40:09.914   547  1563 I audio_hw_utils: audio_extn_utils_send_app_type_cfg PLAYBACK app_type 69937, acdb_dev_id 15, sample_rate 48000
10-05 21:40:09.914   547  1563 D ACDB-LOADER: ACDB -> send_audio_cal, acdb_id = 15, path = 0, app id = 0x11131, sample rate = 48000
10-05 21:40:09.914   547  1563 D ACDB-LOADER: ACDB -> AUDIO_SET_VOL_CAL cal type = 12
10-05 21:40:09.914   547  1563 D ACDB-LOADER: ACDB -> audstrm_cal->cal_type.cal_data.cal_size = 16
10-05 21:40:09.915   547  1563 D ACDB-LOADER: ACDB -> GET_AFE_TOPOLOGY_ID for adcd_id 15, Topology Id 1025e
10-05 21:40:09.915   547  1563 D ACDB-LOADER: ACDB -> send_hw_delay : acdb_id = 15 path = 0
10-05 21:40:09.915   547  1563 D audio_hw_primary: enable_audio_route: apply mixer and update path: low-latency-playback quat_i2s
10-05 21:40:09.915   547  1563 D audio_hw_primary: select_devices: done
10-05 21:40:09.918   547  1563 D msm8974_platform: platform_set_channel_map mixer_ctl_name:Playback Channel Map15
10-05 21:40:09.918   547  1563 D msm8974_platform: platform_set_channel_map: set mapping(1 2 0 0 0 0 0 0) for channel:2
10-05 21:40:09.937 24643 24643 W System  : ClassLoader referenced unknown path: /data/app/com.moez.QKSMS-1/lib/arm64
10-05 21:40:09.954 24643 24643 I MultiDex: VM with version 2.1.0 has multidex support
10-05 21:40:09.954 24643 24643 I MultiDex: install
10-05 21:40:09.954 24643 24643 I MultiDex: VM has multidex support, MultiDex support library is disabled.
10-05 21:40:09.963 24643 24643 D skia    : --- SkImageDecoder::Factory returned null
10-05 21:40:09.963 24643 24643 D skia    : --- SkImageDecoder::Factory returned null
10-05 21:40:09.967 24643 24643 V MmsConfig: tag: bool value: enabledMMS - true
10-05 21:40:09.967 24643 24643 V MmsConfig: tag: int value: defaultSMSMessagesPerThread - 500
10-05 21:40:09.967 24643 24643 V MmsConfig: tag: int value: defaultMMSMessagesPerThread - 50
10-05 21:40:09.967 24643 24643 V MmsConfig: tag: int value: minMessageCountPerThread - 10
10-05 21:40:09.967 24643 24643 V MmsConfig: tag: int value: maxMessageCountPerThread - 5000
10-05 21:40:09.967 24643 24643 V MmsConfig: tag: string value: uaProfUrl - http://www.google.com/oha/rdf/ua-profile-kila.xml
10-05 21:40:09.967 24643 24643 V MmsConfig: tag: int value: recipientLimit - -1
10-05 21:40:09.967 24643 24643 V MmsConfig: tag: bool value: enableMultipartSMS - true
10-05 21:40:09.967 24643 24643 V MmsConfig: tag: bool value: enableSplitSMS - false
10-05 21:40:09.967 24643 24643 V MmsConfig: tag: int value: smsToMmsTextThreshold - -1
10-05 21:40:09.967 24643 24643 V MmsConfig: tag: bool value: enableSlideDuration - true
10-05 21:40:09.967 24643 24643 V MmsConfig: tag: int value: maxMessageTextSize - -1
10-05 21:40:09.967 24643 24643 V MmsConfig: tag: bool value: enableSprintVVM - false
10-05 21:40:09.967 24643 24643 V MmsConfig: tag: string value: mmsc - null
10-05 21:40:09.967 24643 24643 V MmsConfig: tag: string value: mmsproxy - null
10-05 21:40:09.967 24643 24643 V MmsConfig: tag: string value: mmsport - null
10-05 21:40:09.970   547  1563 D audio_hw_primary: start_output_stream: exit
10-05 21:40:10.011 24643 24643 I AppCompatViewInflater: app:theme is now deprecated. Please move to using android:theme instead.
10-05 21:40:10.081 24643 24668 D NotificationManager: Old threads: []
10-05 21:40:10.081 24643 24668 D NotificationManager: New threads: []
10-05 21:40:10.085 24643 24671 D OpenGLRenderer: Use EGL_SWAP_BEHAVIOR_PRESERVED: true
10-05 21:40:10.097 24643 24672 D ConnectivityManager: getMobileDataEnabled()+ subId=1
10-05 21:40:10.098  2523 12318 D PhoneInterfaceManager: [PhoneIntfMgr] getDataEnabled: subId=1 phoneId=0
10-05 21:40:10.098  2523 12318 D PhoneInterfaceManager: [PhoneIntfMgr] getDataEnabled: subId=1 retVal=false
10-05 21:40:10.099 24643 24672 D ConnectivityManager: getMobileDataEnabled()- subId=1 retVal=false
10-05 21:40:10.100 24643 24672 E Utils   : exception thrown
10-05 21:40:10.100 24643 24672 E Utils   : java.lang.NoSuchMethodException: setMobileDataEnabled [boolean]
10-05 21:40:10.100 24643 24672 E Utils   :  at java.lang.Class.getMethod(Class.java:624)
10-05 21:40:10.100 24643 24672 E Utils   :  at java.lang.Class.getDeclaredMethod(Class.java:586)
10-05 21:40:10.100 24643 24672 E Utils   :  at com.moez.QKSMS.mmssms.Utils.setMobileDataEnabled(Utils.java:250)
10-05 21:40:10.100 24643 24672 E Utils   :  at com.android.mms.transaction.TransactionService.onNewIntent(TransactionService.java:237)
10-05 21:40:10.100 24643 24672 E Utils   :  at com.android.mms.transaction.TransactionService$ServiceHandler.handleMessage(TransactionService.java:644)
10-05 21:40:10.100 24643 24672 E Utils   :  at android.os.Handler.dispatchMessage(Handler.java:102)
10-05 21:40:10.100 24643 24672 E Utils   :  at android.os.Looper.loop(Looper.java:148)
10-05 21:40:10.100 24643 24672 E Utils   :  at android.os.HandlerThread.run(HandlerThread.java:61)
10-05 21:40:10.112 24643 24671 I Adreno  : QUALCOMM build                   : cababe8, I8a8f1b7e8f
10-05 21:40:10.112 24643 24671 I Adreno  : Build Date                       : 07/16/16
10-05 21:40:10.112 24643 24671 I Adreno  : OpenGL ES Shader Compiler Version: XE031.09.00.03
10-05 21:40:10.112 24643 24671 I Adreno  : Local Branch                     : 
10-05 21:40:10.112 24643 24671 I Adreno  : Remote Branch                    : 
10-05 21:40:10.112 24643 24671 I Adreno  : Remote Branch                    : 
10-05 21:40:10.112 24643 24671 I Adreno  : Reconstruct Branch               : 
10-05 21:40:10.115 24643 24671 I OpenGLRenderer: Initialized EGL, version 1.4
10-05 21:40:10.165 24643 24654 W CursorWrapperInner: Cursor finalized without prior close()
10-05 21:40:10.261 24643 24667 E Mms     : [839] Tried to add duplicate Conversation to Cache for threadId: 49 new conv: [48XXXXXXXXX] (tid 49)
10-05 21:40:10.276 24643 24643 D ConnectivityManager: getMobileDataEnabled()+ subId=1
10-05 21:40:10.277  2523  4864 D PhoneInterfaceManager: [PhoneIntfMgr] getDataEnabled: subId=1 phoneId=0
10-05 21:40:10.278  2523  4864 D PhoneInterfaceManager: [PhoneIntfMgr] getDataEnabled: subId=1 retVal=false
10-05 21:40:10.278 24643 24643 D ConnectivityManager: getMobileDataEnabled()- subId=1 retVal=false
10-05 21:40:10.294 24643 24643 E Utils   : exception thrown
10-05 21:40:10.294 24643 24643 E Utils   : java.lang.NoSuchMethodException: setMobileDataEnabled [boolean]
10-05 21:40:10.294 24643 24643 E Utils   :  at java.lang.Class.getMethod(Class.java:624)
10-05 21:40:10.294 24643 24643 E Utils   :  at java.lang.Class.getDeclaredMethod(Class.java:586)
10-05 21:40:10.294 24643 24643 E Utils   :  at com.moez.QKSMS.mmssms.Utils.setMobileDataEnabled(Utils.java:250)
10-05 21:40:10.294 24643 24643 E Utils   :  at com.android.mms.transaction.TransactionService.onDestroy(TransactionService.java:455)
10-05 21:40:10.294 24643 24643 E Utils   :  at android.app.ActivityThread.handleStopService(ActivityThread.java:3074)
10-05 21:40:10.294 24643 24643 E Utils   :  at android.app.ActivityThread.-wrap21(ActivityThread.java)
10-05 21:40:10.294 24643 24643 E Utils   :  at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1466)
10-05 21:40:10.294 24643 24643 E Utils   :  at android.os.Handler.dispatchMessage(Handler.java:102)
10-05 21:40:10.294 24643 24643 E Utils   :  at android.os.Looper.loop(Looper.java:148)
10-05 21:40:10.294 24643 24643 E Utils   :  at android.app.ActivityThread.main(ActivityThread.java:5461)
10-05 21:40:10.294 24643 24643 E Utils   :  at java.lang.reflect.Method.invoke(Native Method)
10-05 21:40:10.294 24643 24643 E Utils   :  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
10-05 21:40:10.294 24643 24643 E Utils   :  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
10-05 21:40:10.294 24643 24643 E Utils   :  at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:102)
10-05 21:40:10.313 24643 24643 D BindMenuItemTask: onPostExecute: 0
10-05 21:40:10.369  1359  1591 I ActivityManager: Displayed com.moez.QKSMS/.ui.MainActivity-Default: +469ms
10-05 21:40:11.915   756   860 E ANDR-PERF-OPTSHANDLER: perf_lock_rel: updated /sys/class/scsi_host/host0/../../../clkscale_enable with 1
10-05 21:40:11.915   756   860 E ANDR-PERF-OPTSHANDLER:  return value 2
10-05 21:40:12.611  1359  2387 I ActivityManager: START u0 {cmp=com.moez.QKSMS/.ui.messagelist.MessageListActivity (has extras)} from uid 10082 on display 0
10-05 21:40:12.620   547  1563 D audio_hw_primary: out_set_parameters: enter: usecase(1: low-latency-playback) kvpairs: routing=2
10-05 21:40:12.621   547  1563 D audio_hw_extn: audio_extn_set_anc_parameters: anc_enabled:0
10-05 21:40:12.631 24643 24643 I test    : application is in foreground: false
10-05 21:40:12.656   547  1563 D AudioFlinger: mixer(0xf2300000) throttle end: throttle time(6)
10-05 21:40:12.681 24643 24643 I AppCompatViewInflater: app:theme is now deprecated. Please move to using android:theme instead.
10-05 21:40:12.694 24643 24643 V MessageListActivity: Opening thread: 49
10-05 21:40:12.711 24643 24643 D MessageListFragment: LoadConversationTask
10-05 21:40:12.712 24643 24689 D MessageListFragment: Loading conversation
10-05 21:40:12.795 24643 24643 W View    : requestLayout() improperly called by com.moez.QKSMS.ui.view.MessageListRecyclerView{acb2091 VFED..... ......ID 0,0-1080,1680 #7f1000c5 app:id/conversation} during layout: running second layout pass
10-05 21:40:12.796 24643 24643 D MessageListFragment: Conversation loaded
10-05 21:40:12.849  1359  1591 I ActivityManager: Displayed com.moez.QKSMS/.ui.messagelist.MessageListActivity: +212ms
10-05 21:40:12.862 24643 24643 I WebViewFactory: Loading com.android.webview version 51.0.2704.106 (code 1)
10-05 21:40:12.886 24643 24643 W System  : ClassLoader referenced unknown path: /system/app/webview/lib/arm64
10-05 21:40:12.897 24643 24643 I cr_LibraryLoader: Time to load native libraries: 3 ms (timestamps 5986-5989)
10-05 21:40:12.898 24643 24643 I cr_LibraryLoader: Expected native library version number "51.0.2704.106", actual native library version number "51.0.2704.106"
10-05 21:40:12.904 24643 24643 V WebViewChromiumFactoryProvider: Binding Chromium to main looper Looper (main, tid 1) {9e45001}
10-05 21:40:12.905 24643 24643 I cr_LibraryLoader: Expected native library version number "51.0.2704.106", actual native library version number "51.0.2704.106"
10-05 21:40:12.905 24643 24643 I chromium: [INFO:library_loader_hooks.cc(143)] Chromium logging enabled: level = 0, default verbosity = 0
10-05 21:40:12.915  1359  1705 D WifiService: New client listening to asynchronous messages
10-05 21:40:12.923 24643 24643 I cr_BrowserStartup: Initializing chromium process, singleProcess=true
10-05 21:40:12.971 24643 24702 W cr_media: Requires BLUETOOTH permission
10-05 21:40:12.974 24643 24643 I cr_DRP  : No DRP key due to exception:java.lang.ClassNotFoundException: com.android.webview.chromium.Drp
10-05 21:40:13.019  1359  2387 D ConnectivityService: listenForNetwork for Listen from uid/pid:10082/24643 for NetworkRequest [ id=38, legacyType=-1, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED&NOT_VPN] ]
10-05 21:40:14.313  4282  4282 I TriggerBroadcastReceiver: Broadcast received: android.intent.action.BATTERY_CHANGED
10-05 21:40:14.314  4282  4282 I TriggerService: batterylevel changed to 20
10-05 21:40:14.315  4282  4282 I TriggerService: compareTriggers called
10-05 21:40:14.315  4282  4282 I TriggerService: comparePriorities called
```

**Comment by gkomarnicki** [comment]
I have the same issue; three of my contacts have their numbers displayed instead of their names (in my case without the country code though). In other words, 09xxxxxxxxx rather than +63xxxxxxxxxx. All three numbers are in the +63 country code format on the phone.

**Comment by fungs** [comment]
Same here (on a dual sim phone), contact has the international format but the message sender is the local format and the matching doesn't work. I guess one has to use the country prefix for the recipient sim card to construct the international number for lookup?

**Comment by moezbhatti** [comment]
This should be addressed in QKSMS 3.0

**Comment by fungs** [comment]
Does that mean it is already implemented somewhere? Otherwise, it might be better to assign this to a milestone and tag it correspondingly.

**Comment by moezbhatti** [comment]
At the time that I closed this, there were around 400 open issues and I hadn't managed it in almost 2 years

In an effort to clean up the github issues and organize them, I closed all the issues that I believed QKSMS 3.0 _should_ have solved, given that it was re-written from scratch, and going through and testing every single one would have been unfeasible for me as the only developer and project maintainer.

If you're still experiencing this issue, I'm happy to reopen it and look into it! Otherwise, I expect that it was fixed when I re-wrote the contact management in the app

**Comment by fungs** [comment]
Great work! Just wasn't sure what it should mean. Thx!
